store/rootmulti/store.go

```diff
@@ -827,34 +827,34 @@ loop:
 		if err == io.EOF {
 			break
 		} else if err != nil {
-			return snapshottypes.SnapshotItem{}, sdkerrors.Wrap(err, "invalid protobuf message")
+			return snapshottypes.SnapshotItem{}, errorsmod.Wrap(err, "invalid protobuf message")
 		}
 
 		switch item := snapshotItem.Item.(type) {
 		case *snapshottypes.SnapshotItem_Store:
 			if importer != nil {
 				err = importer.Commit()
 				if err != nil {
-					return snapshottypes.SnapshotItem{}, sdkerrors.Wrap(err, "IAVL commit failed")
+					return snapshottypes.SnapshotItem{}, errorsmod.Wrap(err, "IAVL commit failed")
 				}
 				importer.Close()
 			}
 			store, ok := rs.GetStoreByName(item.Store.Name).(*iavl.Store)
 			if !ok || store == nil {
-				return snapshottypes.SnapshotItem{}, sdkerrors.Wrapf(sdkerrors.ErrLogic, "cannot import into non-IAVL store %q", item.Store.Name)
+				return snapshottypes.SnapshotItem{}, errorsmod.Wrapf(types.ErrLogic, "cannot import into non-IAVL store %q", item.Store.Name)
 			}
 			importer, err = store.Import(int64(height))
 			if err != nil {
-				return snapshottypes.SnapshotItem{}, sdkerrors.Wrap(err, "import failed")
+				return snapshottypes.SnapshotItem{}, errorsmod.Wrap(err, "import failed")
 			}
 			defer importer.Close()
 
 		case *snapshottypes.SnapshotItem_IAVL:
 			if importer == nil {
-				return snapshottypes.SnapshotItem{}, sdkerrors.Wrap(sdkerrors.ErrLogic, "received IAVL node item before store item")
+				return snapshottypes.SnapshotItem{}, errorsmod.Wrap(types.ErrLogic, "received IAVL node item before store item")
```

Comment:
So here is my main question/concern:

Why ever do `sdkerrors.Wrap(err, "...")` over the stdlib `fmt.Errorf("...: %w", err)`? What does `sdkerrors` give you? To me it's just completely unnecessary, no?

Reply:
I was thinking the same, but wanted to see what people thought.

Reply:
I think we cant do this in the current design of errors since we have our own custom error type in the errors package

Reply:
nvm I was wrong

Reply:
Yeah, I don't see any need for `sdkerrors.Wrap`. It only provides a `GRPCStatus` method, which is irrelevant for most use-cases, and can easily be moved to a non-receiver function that takes an error as an argument.

Reply:
The wrapping I believe adds a stack trace and preserves info about the source error. It improves debuggability

Reply:
It doesn't do anything the stdlib doesn't do. The only difference is stracktrace which can be fetched easily.

Reply:
There is value in just having the stack trace as part of it. I don't think the error wrapping is the issue we have here. I'll leave it as is and we can come back to it later.